geo/src/algorithm/scale.rs

```diff
@@ -0,0 +1,38 @@
+use crate::{AffineOps, AffineTransform, BoundingRect, CoordNum, MapCoords, Point};
+
+/// An affine transformation which scales a geometry up or down by a factor.
+///
+/// # Examples
+///
+/// ```
+/// use geo::Scale;
+/// use geo::{line_string, BoundingRect, LineString};
+///
+/// let ls: LineString<f64> = line_string![
+///     (x: 0.0f64, y: 0.0f64),
+///     (x: 10.0f64, y: 10.0f64),
+/// ];
+/// let origin = ls.bounding_rect().unwrap().center();
+/// let scaled = ls.scale(10.0, 10.0, origin.into());
+///
+/// assert_eq!(scaled, line_string![
+///     (x: -45.0f64, y: -45.0f64),
+///     (x: 55.0f64, y: 55.0f64)
+/// ]);
+/// ```
+pub trait Scale<T> {
+    fn scale(&self, x: T, y: T, origin: Point<T>) -> Self
+    where
+        T: CoordNum;
+}
+
+impl<T, G> Scale<T> for G
+where
+    T: CoordNum,
+    G: MapCoords<T, T, Output = G> + BoundingRect<T>,
+{
+    fn scale(&self, x: T, y: T, origin: Point<T>) -> Self {
```

Comment:
Bikeshed:  Immutable vs (eventual) Mutable flavors.

Do we want a mutable version of the affine transformations? (I think so, at least eventually!)

Consider further constraining our trait impl from only `MapCoords` to `MapCoords + MapCoordsInPlace` to avoid a future "breakage" when the mutable methods get added.

And then, what should the naming conventions for mutable vs immutable be? Do the existing names here clash with where will end up once there are mutable flavors?

For context, sometimes we use an `_in_place` suffix to denote mutable - as in:
```rust
// Approach A
trait Rotate {
  // immutable, implicitly clones the input.
  fn translate(&self) -> Self;

  // mutable
  fn translate_in_place(&mut self);
}
```

Other places we use verb for mutable and past participle for immutable:
```rust
// Approach B
trait Transform {
  // immutable, implicitly clones the input.
  fn transformed(&self, proj: &Proj) -> Result<Self::Output, ProjError>;

  // mutable
  fn transform(&mut self, proj: &Proj) -> Result<(), ProjError>;
}
```

I generally prefer approach B (and so does the [top comment in some random reddit thread](https://www.reddit.com/r/rust/comments/p0e95q/naming_convention_for_methods_that_mutate_vs/)!), but I'm also ok with the `_in_place` convention, and we're already using it for `translate_in_place`, so there's no clear best option IMO. 🤷

Reply:
Added `MapCoordsInPlace` trait bound.